modules/common/service/types.go

```diff
@@ -67,3 +110,136 @@ const (
 	// MetalLBLoadBalancerIPs -
 	MetalLBLoadBalancerIPs = "metallb.universe.tf/loadBalancerIPs"
 )
+
+// OverrideSpec - service override configuration for the Service created to serve traffic to the cluster.
+// Allows for the manifest of the created Service to be overwritten with custom configuration.
+type OverrideSpec struct {
+	// EndpointURL to be used to register the service in keystone.
+	// +optional
+	EndpointURL *string `json:"endpointURL,omitempty"`
+
```

Comment:
I think we are mixing two use cases.

1. Expose a service via a Route created by the openstack-operator
    This to work we need both the Service override for the openstack-operator to pass down a label selector to the service operator, and the endpointURL for the openstack-operator to pass down the URL of the created Route.
2. Allow the end user to specify extra configuration for the k8s Service created by a service operator.
    For this we only need the Service override but not the endpointURL

Do we have services that only need use case 2) but not 1) and therefore does no need the endpointURL? 

At least I see that in nova-operator nova-metadata will have a service override (as it will have a k8s Service) but openstack-operator will not create a Route for it (as this API is not for external consumption) and therefore the endpointURL won't ever be set.

Can we decouple the two use cases? The use case 1) could use the current struct while the use case 2) could use a different struct that only has EmbeddedLabelsAnnotations and OverrideServiceSpec fields but not EndpointURL (as that would be unused).

We could even avoid duplicated definition with
```golang
type ServiceOverrideSpec struct {
    *EmbeddedLabelsAnnotations `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
	Spec *OverrideServiceSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
}

type RoutedServiceOverrideSpec struct {
    ServiceOverrideSpec
    EndpointURL *string `json:"endpointURL,omitempty"`
}
```

Reply:
yes that is correct, we have some services which don't require the EndpointURL, without checking its, as you said NovaMetadata, and DNSmasq. maybe there are more, or there will be a new service. I'll update the PR to support that situation.

Reply:
done